### PR TITLE
Support empty party display name in scenario service 

### LINKED ISSUE #11099: allocateParty "" throws an error on 1.17.0
"Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))"

### PR DESCRIPTION
**Pull request by aherrmann-da**
Closes #11099 
Adds a regression test

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by S11001001 — APPROVED** [review]

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala [hunk 1]
```diff
@@ -278,7 +278,9 @@ class IdeLedgerClient(
           }
         } else {
           // Allocate a fresh name based on the display name.
-          val candidates = displayName #:: LazyList.from(1).map(displayName + _.toString())
+          // Empty party ids are not allowed, fall back to "party" on empty display name.
+          val namePrefix = if (displayName.isEmpty) { "party" } else { displayName }
+          val candidates = namePrefix #:: LazyList.from(1).map(namePrefix + _.toString())
```
Comment (stefanobaghino-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:283]: I'm not really sure about this solution. Is this also what happens when allocating on a ledger via Script? Would it make sense to error out on the client with a more telling error?

**Review by stefanobaghino-da — COMMENTED** [review]
I'm not so sure about this.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala (hunk 1 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:283]: Running against sandbox with a script sandbox accepts this and assigns some automatic party identifier.

Comment (stefanobaghino-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:283]: Is it `"party"` or something else?

Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:283]: It generates identifiers of the form `"party-0f47e587", "party-39f0402f"`, independent of the display-name. See [here for participant-state](https://github.com/digital-asset/daml/blob/4dd5663c5b38c4aaf4968325cb902c18bd387d84/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateWriter.scala#L96-L97) and [here for sandbox-classic](https://github.com/digital-asset/daml/blob/4dd5663c5b38c4aaf4968325cb902c18bd387d84/ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/PartyIdGenerator.scala#L12).
The script service is different in that it uses the display-name and appends a consecutive count if the name is already taken.

Comment (stefanobaghino-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala]: Oh, I see, thanks.

**Pull request merged**